supplier/spring-file-supplier/src/main/java/org/springframework/cloud/fn/supplier/file/FileSupplierConfiguration.java

```diff
@@ -90,6 +94,7 @@ else if (this.fileSupplierProperties.getFilenameRegex() != null) {
 	}
 
 	@Bean
+	@ConditionalOnProperty(prefix = "file.supplier", name = "tail", matchIfMissing = true)
```

Comment:
I believe this condition does the same as not having a condition at all. Am I missing something?

Reply:
This condition means `true` if no that `file.supplier.tail` property provided.
Therefore another one you are pointing out is opposite: it will `true` only if present.
So, no need anything more complex.

Reply:
The `matchIfMissing` is more useful when the `value` attribute is set. I believe this actually means "`true` if `file.supplier.tail` is not set at all OR if it is set to anything besides `false`".

Reply:
Well, let's look into the code:
```
		private void collectProperties(PropertyResolver resolver, List<String> missing, List<String> nonMatching) {
			for (String name : this.names) {
				String key = this.prefix + name;
				if (resolver.containsProperty(key)) {
					if (!isMatch(resolver.getProperty(key), this.havingValue)) {
						nonMatching.add(name);
					}
				}
				else {
					if (!this.matchIfMissing) {
						missing.add(name);
					}
				}
			}
		}
```
and then:
```
		if (!missingProperties.isEmpty()) {
			return ConditionOutcome.noMatch(ConditionMessage.forCondition(ConditionalOnProperty.class, spec)
				.didNotFind("property", "properties")
				.items(Style.QUOTE, missingProperties));
		}
		if (!nonMatchingProperties.isEmpty()) {
			return ConditionOutcome.noMatch(ConditionMessage.forCondition(ConditionalOnProperty.class, spec)
				.found("different value in property", "different value in properties")
				.items(Style.QUOTE, nonMatchingProperties));
		}
		return ConditionOutcome
			.match(ConditionMessage.forCondition(ConditionalOnProperty.class, spec).because("matched"));
```
So, since no property and `matchIfMissing == true`, both those collections are going to be empty. Therefore we got to the end where condition is `match`.

Reply:
Run `org.springframework.cloud.fn.supplier.file.LinesPayloadTests` and `org.springframework.cloud.fn.supplier.file.TailModeTests` in debugger and put breakpoint on org.springframework.cloud.fn.supplier.file.FileSupplierConfiguration` lines 112,121,128 and you will find that 112 gets hit every time regardless of property being set or not.

Reply:
Got it!
So, `matchIfMissing = true` means by default, when we don't have property.
Then Javadoc is confusing. 🤷 

Thanks

Reply:
👍🏻 Yeh, the docs are confusing on this one.